### PR TITLE
`Development`: Fix start of Artemis in GitLab/Jenkins setups

### PR DESCRIPTION
**Pull request by b-fein**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context
Artemis 6.3.8 does not start in GitLab setups since the optional new configuration value introduced in #6948 is not truly optional, but required since no default is set.

### Description
Makes the configuration value properly optional.
Gives the attribute a better name, since the token is used to authenticate against GitLab, not Jenkins.

### Steps for Testing
1. Start Artemis in a GitLab/Jenkins setup.
2. Artemis starts.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### CODE REVIEW

**Review by Strohgelaender — APPROVED** [review]

File: src/test/java/de/tum/in/www1/artemis/connector/GitlabRequestMockProvider.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ public class GitlabRequestMockProvider {
     private URL gitlabServerUrl;
 
     @Value("${artemis.version-control.health-api-token}")
```
Comment (Strohgelaender) [inline src/test/java/de/tum/in/www1/artemis/connector/GitlabRequestMockProvider.java:62]:
```suggestion
    @Value("${artemis.version-control.health-api-token:#{null}}")
```

**Review by tobias-lippert — APPROVED** [review]
Code LGTM

**Review by TheZoker — APPROVED** [review]

File: src/test/java/de/tum/in/www1/artemis/connector/GitlabRequestMockProvider.java (hunk 1 above)
Comment (b-fein) [reply src/test/java/de/tum/in/www1/artemis/connector/GitlabRequestMockProvider.java:62]: The test config provides a default value, and the attribute is not specified as `Optional` here. Therefore, I’d prefer to leave it as-is.

**Review by jakubriegel — APPROVED** [review]
Thanks for the fix. Code LGTM

**Pull request merged**